tests/conftest.py

```diff
@@ -1379,24 +1379,38 @@ def collect_db_dump_on_duts(request, duthosts):
     if hasattr(request.node, 'rep_call') and request.node.rep_call.failed:
         dut_file_path = "/tmp/db_dump"
         docker_file_path = "./logs/db_dump"
-        db_dump_path = os.path.join(dut_file_path, request.module.__name__, request.node.name)
         db_dump_tarfile = "{}.tar.gz".format(dut_file_path)
 
         # Collect DB config
         dbs = set()
         result = duthosts[0].shell("cat /var/run/redis/sonic-db/database_config.json")
         db_config = json.loads(result['stdout'])
+        state_db_id = db_config['DATABASES']['STATE_DB']['id']
         for db in db_config['DATABASES']:
             db_id = db_config['DATABASES'][db]['id']
             dbs.add(db_id)
 
-        # Collect DB dump
-        duthosts.file(path = db_dump_path, state="directory")
-        for i in dbs:
-            duthosts.shell("redis-dump -d {} -y -o {}/{}".format(i, db_dump_path, i))
+        namespace_list = duthosts[0].get_asic_namespace_list() if duthosts[0].is_multi_asic else []
+        if namespace_list:
+            for namespace in namespace_list:
+                # Collect DB dump
+                db_dump_path = os.path.join(dut_file_path + "/" + namespace, request.module.__name__, request.node.name)
+                duthosts.file(path=db_dump_path, state="directory")
+                for i in dbs:
+                    if i == state_db_id and duthosts[0].sonic_release in ['201911']:
+                        continue
+                    else:
+                        duthosts.shell("ip netns exec {} redis-ducmp -d {} -y -o {}/{}".format(namespace, i, db_dump_path, i))
+        else:
+            # Collect DB dump
+            db_dump_path = os.path.join(dut_file_path, request.module.__name__, request.node.name)
+            duthosts.file(path = db_dump_path, state="directory")
+            for i in dbs:
```

Comment:
`STATEDB` is not skipped here. The better place is not to skip it in line 1400. It would be easier if the STATEDB id is removed from the list `dbs`.
BTW, when skipping STATEDB, it would be better to explain the reason in comments.

Reply:
OK, I'll change it